x/concentrated-liquidity/tick.go

```diff
@@ -253,6 +253,68 @@ func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]quer
 	return liquidityDepthsForRange, nil
 }
 
+// GetLiquidityNetInDirection returns an array of liquidity depth in the given zeroForOne direction.
+// Uses boundTick if given, and iterates until we hit bound Tick's index. If not, uses min tick || max tick depending on the direction of zeroForOne.
+func (k Keeper) GetLiquidityNetInDirection(ctx sdk.Context, poolId uint64, zeroForOne bool, boundTickPointer *sdk.Int) ([]query.LiquidityDepth, error) {
+	// check if pool exists
+	if !k.poolExists(ctx, poolId) {
+		return []query.LiquidityDepth{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+
+	// get min and max tick for the pool
+	p, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return []query.LiquidityDepth{}, err
+	}
+	currentTick := p.GetCurrentTick()
+
+	// use max or min tick if provided bound is nil
+	var boundTick sdk.Int
+	if boundTickPointer == nil {
+		exponentAtPriceOne := p.GetPrecisionFactorAtPriceOne()
+		minTick, maxTick := math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
+
+		if zeroForOne {
+			boundTick = sdk.NewInt(minTick)
+		} else {
+			boundTick = sdk.NewInt(maxTick)
+		}
+	} else { // if bound is not nil, sanity check that given bound is in the direction of swap
+		if (zeroForOne && boundTick.LT(currentTick)) || (!zeroForOne && boundTick.GT(currentTick)) {
+			return []query.LiquidityDepth{}, types.InvalidDirectionError{ZeroForOne: zeroForOne, PoolTick: currentTick.Int64(), TargetTick: boundTick.Int64()}
+		}
+		boundTick = *boundTickPointer
+	}
+
+	liquidityDepths := []query.LiquidityDepth{}
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey, sdk.ZeroDec())
+
+	for (currentTick.GT(boundTick) && zeroForOne) || (currentTick.LT(boundTick) && !zeroForOne) {
```

Comment:
Can we expand https://github.com/osmosis-labs/osmosis/issues/4704 to also optimize this with an iterator?

I'm also guessing that it should be possible to share some of the iteration logic with "query for liquidity depths"?

Reply:
yes of course! I think in #4707, we should be addressing both queries. 
Addinng it to the issue description as well